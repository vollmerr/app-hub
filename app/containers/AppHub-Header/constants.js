/*
 *
 * AppHubHeader constants
 *
 */

export const DEFAULT_ACTION = 'app/AppHubHeader/DEFAULT_ACTION';
