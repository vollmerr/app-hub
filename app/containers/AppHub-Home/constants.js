/*
 *
 * AppHubHome constants
 *
 */

export const DEFAULT_ACTION = 'app/AppHubHome/DEFAULT_ACTION';
