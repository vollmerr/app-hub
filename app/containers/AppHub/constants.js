// VIEW
export const CHANGE_MOBILE = 'AppHub/CHANGE_MOBILE';
export const CHANGE_PANEL_OPEN = 'AppHub/CHANGE_PANEL_OPEN';
export const CHANGE_PANEL_SELECTED = 'AppHub/CHANGE_PANEL_SELECTED';
// panel names
export const HELP_PANEL = 'AppHub/HELP_PANEL';
export const APPS_PANEL = 'AppHub/APPS_PANEL';
export const APP_NAV_PANEL = 'AppHub/APP_NAV_PANEL';
export const ALERTS_PANEL = 'AppHub/ALERTS_PANEL';
export const DEV_PANEL = 'AppHub/DEV_PANEL';
// APP
export const CHANGE_APP = 'AppHub/CHANGE_APP';
export const CHANGE_APP_STATUS = 'AppHub/CHANGE_APP_STATUS';
// user
export const AUTH_USER = 'AppHub/AUTH_USER';
export const AUTH_USER_DONE = 'AppHub/AUTH_USER_DONE';
// REQUEST STATUS TYPES
export const REQUEST = '_REQUEST';
export const SUCCESS = '_SUCCESS';
export const FAILURE = '_FAILURE';

export const COL_TYPES = {
  DATE: 'DATE',
};

export const ROLES = {
  USER: 'AppHub User',
};

// D3
export const INCREMENT_RENDER_COUNT = 'AppHub/INCREMENT_RENDER_COUNT';
