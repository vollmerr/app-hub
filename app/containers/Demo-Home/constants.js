export const EXAMPLE_DATA_REQUEST = 'app/DemoHome/EXAMPLE_DATA_REQUEST';
export const EXAMPLE_DATA_SUCCESS = 'app/DemoHome/EXAMPLE_DATA_SUCCESS';
export const EXAMPLE_DATA_FAILURE = 'app/DemoHome/EXAMPLE_DATA_FAILURE';
