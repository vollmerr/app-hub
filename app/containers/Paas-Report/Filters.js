import React from 'react';
// import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Dropdown } from 'office-ui-fabric-react/lib/Dropdown';

import theme from 'utils/theme';


export const Wrapper = styled.div`
  flex: 2 0 50vh;
  margin: 5px;
  padding: 15px;
  background: ${theme.white};
  box-shadow: 0 0 3px ${theme.neutralLight};
  overflow: auto;
`;


// export const Heading = styled.h3`
//   margin: 0 5px 10px 5px;
// `;


// export const Item = styled.div`
//   display: inline-flex;
//   width: 100%;
//   padding: 5px;
// `;

// export const Key = styled.div`
//   flex: 1;
//   padding-right: 10px;
// `;


// export const Value = styled.div`
//   flex: 3;
// `;


// export const items = [
//   ACK.STATUS,
//   ACK.START_DATE,
//   ACK.END_DATE,
//   ACK.TARGET_GROUPS,
//   ACK.FILE_NAME,
//   ACK.STATEMENT,
//   ACK.DETAILS,
// ];

class Filters extends React.PureComponent {
  render() {
    // const { selectedItem, enums } = this.props;

    return (
      <Wrapper>
        {/* <Heading>{selectedItem[ACK.TITLE]}</Heading>
        {
          items.map((name) => (
            <Item key={name}>
              <Key>{acknowledgment[name].label}</Key>
              <Value>{formatItem(selectedItem, name, acknowledgment[name], enums)}</Value>
            </Item>
          ))
        } */}
        <Dropdown
          label={'Report Year'}
          options={[
            { key: '2017', text: '2017' },
            { key: '2016', text: '2016' },
            { key: '2015', text: '2015' },
            { key: '2014', text: '2014' },
          ]}
        />
      </Wrapper>
    );
  }
}


// const { object } = PropTypes;

Filters.propTypes = {

};

export default Filters;
