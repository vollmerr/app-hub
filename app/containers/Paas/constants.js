export const GET_MANAGER_DATA_REQUEST = 'app/Paas/GET_MANAGER_DATA_REQUEST';
export const GET_MANAGER_DATA_SUCCESS = 'app/Paas/GET_MANAGER_DATA_SUCCESS';
export const GET_MANAGER_DATA_FAILURE = 'app/Paas/GET_MANAGER_DATA_FAILURE';

export const UPDATE_USERS_REQUEST = 'app/Paas/UPDATE_USERS_REQUEST';
export const UPDATE_USERS_SUCCESS = 'app/Paas/UPDATE_USERS_SUCCESS';
export const UPDATE_USERS_FAILURE = 'app/Paas/UPDATE_USERS_FAILURE';

export const GET_REPORT_DATA_REQUEST = 'app/Paas/GET_REPORT_DATA_REQUEST';
export const GET_REPORT_DATA_SUCCESS = 'app/Paas/GET_REPORT_DATA_SUCCESS';
export const GET_REPORT_DATA_FAILURE = 'app/Paas/GET_REPORT_DATA_FAILURE';

export const SET_REPORT_KEY = 'app/Paas/SET_REPORT_KEY';
export const SET_REPORT_FILTER = 'app/Paas/SET_REPORT_FILTER';

// ROLES
export const ROLES = {
  MANAGER: 'PAAS Manager',
  SECURITY: 'PAAS Security',
  HR: 'PAAS HR',
};
export const ROLES_VALUES = Object.values(ROLES);
// APPROVAL CODES
export const APPROVAL = {
  NONE: undefined,
  DENY: 0,
  APPROVE: 1,
};
// ACCESS AUTHORIZATION
export const AUTH = {
  ID: '_id',
  SID: 'sid',
  FULL_NAME: 'fullName',
  EMAIL: 'email',
  POS_NUMBER: 'posNumber',
  MANAGER_SID: 'managerSid',
  MANAGER_NAME: 'managerFullName',
  STATUS: 'status',
  LAST_MODIFIED: 'lastUpdated',
  LAST_APPROVED: 'lastApproved',
  DATE_CREATED: 'created',
  AUTH_YEAR: 'authYear',
};
// AUTH STATUS
export const STATUS = {
  ACTIVE: 'active',
  INACTIVE: 'inactive',
  NO_MANAGER: 'noManager',
};
// APP TO AUTHORIZE
export const APPS = {
  APP_1: 'app1',
  APP_2: 'app2',
  APP_3: 'app3',
  APP_4: 'app4',
};
export const APP_LIST = Object.values(APPS);
export const BUTTONS = {
  APPROVE: 'approveButton',
  DENY: 'denyButton',
};
export const REPORT = {
  APPROVED: 0,
  DENIED: 1,
  PENDING: 2,
  NO_MANAGER: 3,
};
