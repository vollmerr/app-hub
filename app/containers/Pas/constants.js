export const GET_MANAGER_DATA_REQUEST = 'app/Pas/GET_MANAGER_DATA_REQUEST';
export const GET_MANAGER_DATA_SUCCESS = 'app/Pas/GET_MANAGER_DATA_SUCCESS';
export const GET_MANAGER_DATA_FAILURE = 'app/Pas/GET_MANAGER_DATA_FAILURE';

export const UPDATE_USERS_REQUEST = 'app/Pas/UPDATE_USERS_REQUEST';
export const UPDATE_USERS_SUCCESS = 'app/Pas/UPDATE_USERS_SUCCESS';
export const UPDATE_USERS_FAILURE = 'app/Pas/UPDATE_USERS_FAILURE';

export const GET_REPORT_DATA_REQUEST = 'app/Pas/GET_REPORT_DATA_REQUEST';
export const GET_REPORT_DATA_SUCCESS = 'app/Pas/GET_REPORT_DATA_SUCCESS';
export const GET_REPORT_DATA_FAILURE = 'app/Pas/GET_REPORT_DATA_FAILURE';

export const SET_REPORT_KEY = 'app/Pas/SET_REPORT_KEY';
export const SET_REPORT_FILTER = 'app/Pas/SET_REPORT_FILTER';

export const UPDATE_USER_MANAGER_REQUEST = 'app/Pas/UPDATE_USER_MANAGER_REQUEST';
export const UPDATE_USER_MANAGER_SUCCESS = 'app/Pas/UPDATE_USER_MANAGER_SUCCESS';
export const UPDATE_USER_MANAGER_FAILURE = 'app/Pas/UPDATE_USER_MANAGER_FAILURE';


// ROLES
export const ROLES = {
  MANAGER: 'PAAS Manager',
  SECURITY: 'PAAS Security',
  HR: 'PAAS HR',
};
export const ROLES_VALUES = Object.values(ROLES);
// APPROVAL CODES
export const APPROVAL = {
  NONE: undefined,
  DENY: 0,
  APPROVE: 1,
};
// ACCESS AUTHORIZATION
export const AUTH = {
  ID: '_id',
  SID: 'sid',
  FULL_NAME: 'fullName',
  EMAIL: 'email',
  POS_NUMBER: 'posNumber',
  MANAGER_SID: 'managerSID',
  MANAGER_NAME: 'managerFullName',
  STATUS: 'status',
  LAST_MODIFIED: 'lastUpdated',
  LAST_APPROVED: 'lastApproved',
  DATE_CREATED: 'created',
  AUTH_YEAR: 'authYear',
};
// AUTH STATUS
export const STATUS = {
  ACTIVE: 'active',
  INACTIVE: 'inactive',
  NO_MANAGER: 'noManager',
  ASSIGNED_MANAGER: 'assignedManager',
};
// APP TO AUTHORIZE
export const APPS = {
  APP_1: 'app1',
  APP_2: 'app2',
  APP_3: 'app3',
  APP_4: 'app4',
};
export const APP_LIST = Object.values(APPS);
export const BUTTONS = {
  APPROVE: 'approveButton',
  DENY: 'denyButton',
};
export const REPORT = {
  APPROVED: 0,
  DENIED: 1,
  PENDING: 2,
  NO_MANAGER: 3,
};

export const MANAGE = {
  EMPLOYEE_ID: 'employee_id',
  MANAGER_ID: 'manager_id',
};
