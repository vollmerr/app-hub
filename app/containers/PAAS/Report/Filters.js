import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Dropdown } from 'office-ui-fabric-react/lib/Dropdown';
import { DefaultButton } from 'office-ui-fabric-react/lib/Button';
import sortBy from 'lodash/sortBy';
// import json2csv from 'json2csv';

import theme from '../../../utils/theme';
// import { formattedDate } from '../../../utils/date';
// import { formatList, downloadFile } from '../../../utils/data';

// import { reportCsv } from '../data';
import * as C from '../constants';


export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  flex: 1 0 auto;
  margin: 5px;
  padding: 15px;
  background: ${theme.white};
  box-shadow: 0 0 3px ${theme.neutralLight};
  overflow: auto;
  > button {
    align-self: center;
  }
`;


class Filters extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      options: {
        [C.AUTH.AUTH_YEAR]: [{ key: 'ALL', text: 'All' }],
      },
    };
  }

  componentDidMount() {
    this.buildOptions();
  }

  /**
   * Builds filter options based off available values
   * from the `allItems` array of items
   */
  buildOptions = () => {
    const { data } = this.props;
    const options = {
      [C.AUTH.AUTH_YEAR]: [],
    };
    const found = {
      [C.AUTH.AUTH_YEAR]: {},
    };
    let key;
    // for all authorization items
    data.forEach((item) => {
      // for all the filter options
      Object.keys(options).forEach((option) => {
        key = item[option];
        // if the filters value hasnt been found for this option
        if (!found[option][key]) {
          // mark as found and add to options
          found[option][key] = 1;
          options[option].push({
            key,
            text: key,
          });
        }
      });
    });
    // sort then add the 'ALL' option to each filter as the first option
    Object.keys(options).forEach((option) => {
      options[option] = sortBy(options[option], 'key').reverse();
      options[option].unshift({ key: 'ALL', text: 'All' });
    });
    // update...
    this.setState({ options });
  }

  // /**
  //  * Handles downloading the report as a csv
  //  */
  // handleDownload = () => {
  //   const { authorizations, selectedKey } = this.props;
  //   const data = formatList(authorizations[selectedKey], reportCsv);
  //   const csv = json2csv({ data, newLine: '\r\n', ...reportCsv });
  //   const name = `PAAS Report ${formattedDate(new Date())}.csv`;
  //   const type = 'data:text/csv;charset=utf-8;';

  //   downloadFile(csv, name, type);
  // }

  render() {
    const { onChange } = this.props;
    const { options } = this.state;

    return (
      <Wrapper>
        <Dropdown
          defaultSelectedKey={'ALL'}
          label={'Report Year'}
          options={options[C.AUTH.AUTH_YEAR]}
          onChanged={onChange(C.AUTH.AUTH_YEAR)}
        />
        <DefaultButton
          iconProps={{ iconName: 'download' }}
          text={'Download CSV'}
          onClick={this.handleDownload}
        />
      </Wrapper>
    );
  }
}


const { array, func } = PropTypes;

Filters.propTypes = {
  data: array,
  onChange: func.isRequired,
};


export default Filters;
