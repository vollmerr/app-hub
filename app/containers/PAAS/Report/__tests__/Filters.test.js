import React from 'react';
import { shallow } from 'enzyme';

import { testStyledComponent } from '../../../../utils/testUtils';

import * as C from '../../constants';

import Filters, { Wrapper } from '../Filters';


testStyledComponent(Wrapper);


const props = {
  data: [
    { title: 'second', [C.AUTH.AUTH_YEAR]: 2003 },
    { title: 'first', [C.AUTH.AUTH_YEAR]: 2007 },
    { title: 'last', [C.AUTH.AUTH_YEAR]: 2001 },
  ],
  onChange: jest.fn(),
};


describe('Filters', () => {
  let wrapper;
  let instance;
  beforeEach(() => {
    wrapper = shallow(<Filters {...props} />);
    instance = wrapper.instance();
  });


  describe('componentDidMount', () => {
    it('should build the options', () => {
      instance.buildOptions = jest.fn();
      instance.componentDidMount();
      expect(instance.buildOptions).toHaveBeenCalled();
    });
  });


  describe('buildOptions', () => {
    it('should build sorted options then update local state', () => {
      instance.buildOptions();
      const options = wrapper.state('options')[C.AUTH.AUTH_YEAR];
      expect(options[0].key).toEqual('ALL');
      expect(options[1].key).toEqual(props.data[1][C.AUTH.AUTH_YEAR]);
      expect(options[3].key).toEqual(props.data[2][C.AUTH.AUTH_YEAR]);
    });

    it('should create unique values', () => {
      const data = [
        ...props.data,
        { title: 'second duplicate', [C.AUTH.AUTH_YEAR]: 2003 },
      ];
      wrapper.setProps({ data });
      instance.buildOptions();
      const options = wrapper.state('options')[C.AUTH.AUTH_YEAR];
      expect(options[0].key).toEqual('ALL');
      expect(options[1].key).toEqual(props.data[1][C.AUTH.AUTH_YEAR]);
      expect(options[3].key).toEqual(props.data[2][C.AUTH.AUTH_YEAR]);
    });
  });


  xdescribe('handleDownload', () => {
    it('should be tested...', () => {
      // TODO: tests
    });
  });


  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
