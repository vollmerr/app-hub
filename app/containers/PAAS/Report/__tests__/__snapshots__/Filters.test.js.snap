// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters__Wrapper /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  margin: 5px;
  padding: 15px;
  background: #ffffff;
  box-shadow: 0 0 3px #eaeaea;
  overflow: auto;
}

.c0 > button {
  -ms-flex-item-align: center;
  align-self: center;
}

<div
  className="c0"
/>
`;

exports[`Filters render should render correctly 1`] = `
<Filters__Wrapper>
  <WithResponsiveMode
    defaultSelectedKey="ALL"
    label="Authorization Year"
    options={
      Array [
        Object {
          "key": "ALL",
          "text": "All",
        },
        Object {
          "key": 2007,
          "text": 2007,
        },
        Object {
          "key": 2003,
          "text": 2003,
        },
        Object {
          "key": 2001,
          "text": 2001,
        },
      ]
    }
  />
  <CustomizedDefaultButton
    iconProps={
      Object {
        "iconName": "download",
      }
    }
    onClick={[Function]}
    text="Download CSV"
  />
</Filters__Wrapper>
`;
